### PR TITLE
Remove assert with error message when rescan web external table

### PR DESCRIPTION
**Pull request by yydzero**
Use better error message instead of assert failure.

### CODE REVIEW

**Comment by vraghavan78** [comment]
+1

**Comment by atris** [comment]
+1

File: src/backend/access/external/url.c [hunk 1]
```diff
@@ -2492,12 +2492,7 @@ url_rewind(URL_FILE *file, const char *relname)
 			break;
 
 		case CFTYPE_EXEC:
-			/* we'll need to execute the command again */
-			assert(0); /* There is no way the following code is right. */
-			url_fclose(file, true, relname);
-
-			/* most of these are null fo us. */
-			url_fopen(url, false, NULL, NULL, NULL, 0);
+			elog(ERROR, "Rescan is not supported for web external table: %s", relname);
```
Comment (danielgustafsson) [inline src/backend/access/external/url.c]: The wording is reversed judging by prior usage in the code and in the documentation, should be "external web table"

Comment (yydzero) [inline src/backend/access/external/url.c]: Thanks for the review, Daniel!

Sounds like we are mixed using 'external web table' and 'web external
table'.  From official GPDB Admin guide (http://docs.pivotal.io/), the
usage is 'web external table'.

On Wed, Mar 9, 2016 at 3:14 PM, Daniel Gustafsson <notifications@github.com>
wrote:

> In src/backend/access/external/url.c
> <https://github.com/greenplum-db/gpdb/pull/489#discussion_r55479567>:
>
> > @@ -2492,12 +2492,7 @@ url_rewind(URL_FILE *file, const char *relname)
> >  			break;
> >
> >  		case CFTYPE_EXEC:
> > -			/* we'll need to execute the command again */
> > -			assert(0); /* There is no way the following code is right. */
> > -			url_fclose(file, true, relname);
> > -
> > -			/* most of these are null fo us. */
> > -			url_fopen(url, false, NULL, NULL, NULL, 0);
> > +			elog(ERROR, "Rescan is not supported for web external table: %s", relname);
>
> The wording is reversed judging by prior usage in the code and in the
> documentation, should be "external web table"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/greenplum-db/gpdb/pull/489/files#r55479567>.
>



-- 
Best Regards,
Yandong

Comment (danielgustafsson) [inline src/backend/access/external/url.c]: We are indeed mixing up the nomenclature in the documentation, we have these articles in the current doc set: [Defining a Command-Based Writable External Web Table](http://gpdb.docs.pivotal.io/4350/admin_guide/load/topics/g-defining-a-command-based-writable-external-web-table.html) and [Command-based Web External Tables](http://gpdb.docs.pivotal.io/4360/admin_guide/load/topics/g-command-based-web-external-tables.html). I guess this needs to be punted to the documentation team to decide on.

Comment (mkiyama) [inline src/backend/access/external/url.c]: FYI - I have submitted a request to get the term standardized.
I suggested using the term "writable external web table" to match the command name.

Comment (danielgustafsson) [inline src/backend/access/external/url.c]: Excellent, thanks for picking this up! I had a mental note to involve you on this but hadn't gotten that far on the TODO list for today.

**Pull request closed without merging**